core/src/main/kotlin/rys/ajaxpetproject/repository/impl/MessageReactiveRepository.kt

```diff
@@ -0,0 +1,67 @@
+package rys.ajaxpetproject.repository.impl
+
+import org.bson.types.ObjectId
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate
+import org.springframework.data.mongodb.core.FindAndModifyOptions
+import org.springframework.data.mongodb.core.findAndModify
+import org.springframework.data.mongodb.core.findById
+import org.springframework.data.mongodb.core.find
+import org.springframework.data.mongodb.core.remove
+import org.springframework.data.mongodb.core.findAll
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+import org.springframework.data.mongodb.core.query.Update
+import org.springframework.stereotype.Repository
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.toMono
+import rys.ajaxpetproject.model.MongoMessage
+import rys.ajaxpetproject.model.MongoUser
+import rys.ajaxpetproject.repository.MessageRepository
+
+@Repository
+class MessageReactiveRepository(private val mongoTemplate : ReactiveMongoTemplate) : MessageRepository {
+    override fun findMessageById(id: ObjectId): Mono<MongoMessage> {
+        val query = Query.query(Criteria.where("id").`is`(id))
+        return mongoTemplate.findById<MongoMessage>(query)
+    }
+
+    override fun save(message: MongoMessage): Mono<MongoMessage> {
+        return mongoTemplate.save(message)
+    }
+
+    override fun deleteAll(): Mono<Unit> {
+        return mongoTemplate.remove<MongoUser>(Query())
+            .flatMap { Unit.toMono()
+            }
+    }
+
+    override fun update(id: ObjectId, message: MongoMessage): Mono<MongoMessage> {
+        val query = Query.query(Criteria.where("id").`is`(id))
+        val updatedMessage = message.copy(id = id)
```

Comment:
why do you need updatedMessage?

Reply:
This is part of Application. I thought it will be more easy to review few small Pull Requests instead of one big. 
All methods here used.
Do I need to make 1 **Big**  PR ? 
@ajax-shulepa-a

Reply:
If it'll be changed later than you can leave as is